sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/trees/TreeNode.scala

```diff
@@ -419,35 +426,77 @@ abstract class TreeNode[BaseType <: TreeNode[BaseType]] extends Product {
    */
   def nodeName: String = getClass.getSimpleName.replaceAll("Exec$", "")
 
+  /** The argument list to be displayed in tree String. */
+  protected def argumentsForStringOutput(verbose: Boolean): Seq[(String, Any)] = {
```

Comment:
Basically if we want this to make it into 2.0, it is not great to do structural changes to treenode/queryplan.